lib/Util/TypeInfo.cpp

```diff
@@ -11,9 +21,45 @@
 
 std::string Pothos::Util::typeInfoToString(const std::type_info &type)
 {
-    //Since std::string is used a lot and often has a complicated template name,
-    //we just enforce returning a simple display name for the std::string type.
-    if (type == typeid(std::string)) return "std::string";
+    #define IfTypeReturnNiceName(T) \
+        if(type == typeid(T)) return #T;
+
+    #define IfTypeContainerReturnNiceName(T) \
+        IfTypeReturnNiceName(std::set<T>) \
+        IfTypeReturnNiceName(std::vector<T>) \
+        IfTypeReturnNiceName(std::vector<std::vector<T>>)
+
+    #define IfTypeNumericContainerReturnNiceName(T) \
+        IfTypeContainerReturnNiceName(T) \
+        IfTypeContainerReturnNiceName(std::complex<T>)
+
+    IfTypeReturnNiceName(Pothos::ObjectKwargs)
+    IfTypeReturnNiceName(Pothos::ObjectMap)
+    IfTypeReturnNiceName(Pothos::ObjectSet)
+    IfTypeReturnNiceName(Pothos::ObjectVector)
+    IfTypeReturnNiceName(Pothos::ProxyMap)
+    IfTypeReturnNiceName(Pothos::ProxySet)
+    IfTypeReturnNiceName(Pothos::ProxyVector)
+    IfTypeReturnNiceName(Pothos::ProxyEnvironment::Sptr)
+    IfTypeReturnNiceName(Pothos::ProxyEnvironmentArgs)
+    IfTypeReturnNiceName(Pothos::ProxyConvertPair)
+
+    IfTypeContainerReturnNiceName(std::string)
+
+    IfTypeNumericContainerReturnNiceName(bool)
+    IfTypeNumericContainerReturnNiceName(char)
+    IfTypeNumericContainerReturnNiceName(signed char)
+    IfTypeNumericContainerReturnNiceName(short)
+    IfTypeNumericContainerReturnNiceName(int)
+    IfTypeNumericContainerReturnNiceName(long)
+    IfTypeNumericContainerReturnNiceName(long long)
+    IfTypeNumericContainerReturnNiceName(unsigned char)
+    IfTypeNumericContainerReturnNiceName(unsigned short)
+    IfTypeNumericContainerReturnNiceName(unsigned int)
+    IfTypeNumericContainerReturnNiceName(unsigned long)
+    IfTypeNumericContainerReturnNiceName(unsigned long long)
+    IfTypeNumericContainerReturnNiceName(float)
+    IfTypeNumericContainerReturnNiceName(double)
```

Comment:
This has got to be a missed opportunity of a hash map. The type_infos all have a hash, we can log2(N) or whatever lookup the type to see if a string is present for it.

Reply:
Done